generated/nitclk/session.py

```diff
@@ -0,0 +1,1156 @@
+    triggers, configure_for_homogeneous_triggers does not affect script
+    triggers. Script triggers are considered to be configured for all
+    sessions if either one or the other of the following conditions are
+    true: - No session has a script trigger that is None - One session has a
+    script trigger that is None and all other sessions have script triggers
+    other than None. The one session with the None trigger must have
+    script_trigger_master_session set to itself, indicating
+    that the session itself is the script trigger master Pause Triggers
+    configure_for_homogeneous_triggers configures generation sessions
+    that support pause triggers to share them, if the pause triggers are
+    None (no trigger configured) for all except one session. The pause
+    triggers are shared by: - Implicitly exporting the pause trigger from
+    the session whose script trigger is not None - Configuring the other
+    sessions that support the pause trigger for digital-edge pause triggers
+    with sources corresponding to the exported pause trigger - Setting
+    pause_trigger_master_session to the session that is
+    exporting the trigger for all sessions that support script triggers If
+    the pause triggers are configured for all generation sessions that
+    support pause triggers, configure_for_homogeneous_triggers does not
+    affect pause triggers. Pause triggers are considered to be configured
+    for all sessions if either one or the other of the following conditions
+    is true: - No session has a pause trigger that is None - One session has
+    a pause trigger that is None and all other sessions have pause triggers
+    other than None. The one session with the None trigger must have
+    pause_trigger_master_session set to itself, indicating
+    that the session itself is the pause trigger master Note: TClk
+    synchronization is not supported for pause triggers on acquisition
+    sessions.
+
+    Args:
+        sessions (list of int): sessions is an array of sessions that are being synchronized.
+
+    '''
+    return _get_session_class().configure_for_homogeneous_triggers(sessions)
+
+
+def finish_sync_pulse_sender_synchronize(sessions, min_time):
+    '''finish_sync_pulse_sender_synchronize
+
+    TBD
+
+    Args:
+        sessions (list of int): sessions is an array of sessions that are being synchronized.
+
+        min_time (float): Minimal period of TClk, expressed in seconds. Supported values are
+            between 0.0 s and 0.050 s (50 ms). Minimal period for a single
+            chassis/PC is 200 ns. If the specified value is less than 200 ns,
+            NI-TClk automatically coerces minTime to 200 ns. For multichassis
+            synchronization, adjust this value to account for propagation delays
+            through the various devices and cables.
+
+    '''
+    return _get_session_class().finish_sync_pulse_sender_synchronize(sessions, min_time)
+
+
+def init_for_documentation(self):
```

Comment:
I'm confused why the methods show up twice in this file.

Reply:
It is an implementation detail: From #1005 

> Internally, there is a class that contains the actual implementation of the functions. The main purpose of this is to allow reusing the default function template as is. Not doing this would require either a separate template whose only difference is indentation, or adding variable indentation to the existing template which would make it very messy. @marcoskirsch didn't like variable indenting when originally attempted for something in the pre 1.0 time frame.

Reply:
So the production code is made more complex and more unlike what a human would directly write in order to simplify the templates?

Reply:
Correct. There is one extra level of indirection in order to reuse templates.